app/jobs/organization_event_job.rb

```diff
@@ -19,20 +19,28 @@ def perform(payload_body)
 
     return true unless @user
 
-    transfer_assignments if user_owns_assignments?
+    transfer_assignments if user_owns_any_assignments?
     @organization.users.delete(@user)
   end
   # rubocop:enable Metrics/AbcSize
 
+  def user_owns_any_assignments?
+    user_owns_assignments? || user_owns_group_assignments?
+  end
+
   def user_owns_assignments?
-    @organization.all_assignments.map(&:creator_id).include? @user.id
+    @organization.assignments.where(creator_id: @user.id).count.positive?
```

Comment:
Can we do `@organization.assignments.where(creator_id: @user.id).any?` here? This skips counting all the records by applying a `LIMIT 1` to the query.

Reply:
A quick test in Rails:

```ruby
  irb(main):003:0> User.all.any?
  User Exists (0.4ms)  SELECT  1 AS one FROM "users" LIMIT ?  [["LIMIT", 1]]
```